src/backend/distributed/citus--8.3-1--8.4-1.sql

```diff
@@ -21,6 +93,8 @@ BEGIN
     -- enterprise catalog tables
     CREATE TABLE public.pg_dist_authinfo AS SELECT * FROM pg_catalog.pg_dist_authinfo;
     CREATE TABLE public.pg_dist_poolinfo AS SELECT * FROM pg_catalog.pg_dist_poolinfo;
+
+    PERFORM citus_internal.prepare_pg_upgrade_pg_dist_object();
```

Comment:
I was kind of expecting us to do something like `UPDATE citus.pg_dist_object SET identifier = get_object_identifier(classid, objid);`

Reply:
I think `citus_prepare_pg_upgrade_pg_dist_object ` does it with `C` :)  We could actually remove `getObjectIdentity ` if we remove C implementation of `prepare_pg_upgrade_pg_dist_object `.

I guess keeping `prepare_pg_upgrade_pg_dist_object ` as a wrapper could be nice because we might end up with doing some extra things in the future?

Reply:
This could definitely have been a SQL statement, the restore needs some function because I couldn't find a postgres function to get from the identity to the oid. So we either have a function that can resolve that, or we keep the `finish` function as is.

Reply:
We don't have to do it, but can't we write `get_object_identifier ()` ? (asking to make sure not to miss anything)

Reply:
I can't find a `get_object_identifier` function in either postgres or citus. But this could be rewritten with `pg_identify_object` I guess.

Benefit is that we have less UDF's to define, which I guess is what we want.

Reply:
fixed, used sql for both `prepare` and `finish`, only 1 simple udf exposed to go from `classid` and `identifier` to `objid`.